qipr_approver/approver/decorators/login_required.py

```diff
@@ -10,7 +10,11 @@ def login_required(view_function):
     """
     def wrapped_view(request, *args, **kwargs):
         if request.session.get(SESSION_VARS.get('gatorlink')):
-            return view_function(request, *args, **kwargs)
-        else:
-            return redirect(reverse("approver:shib"))
+            if __shib_validated(request):
+                return view_function(request, *args, **kwargs)
+        return redirect(reverse("approver:shib"))
+
+    def __shib_validated(request):
+        return ((SHIB_ENABLED == 'true' and request.META.get('HTTP_EPPN')) or (SHIB_ENABLED == 'false'))
```

Comment:
This looks weird. So if shib is enabled we want this HTTP_EPPN thing but also the request if shib validated if the shib isnt enabled? This could use a comment describing why it is like this.